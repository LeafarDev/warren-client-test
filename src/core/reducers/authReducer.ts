import * as actions from '../actionTypes/authActionsTypes';

export interface AuthState {
  token: string;
}

const initialState: AuthState = {
  token: ''
};

export default function authReducer(
  state: AuthState = initialState,
  action: actions.AuthAction
): AuthState {
  switch (action.type) {
    case actions.RESET:
    case actions.LOGOUT:
      return initialState;
    case actions.LOGIN_SUCCESS:
      return {
        token: action.token
      };
    default:
      return state;
  }
}
