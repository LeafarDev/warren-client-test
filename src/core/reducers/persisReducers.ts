import storage from 'redux-persist/lib/storage';
import { persistReducer } from 'redux-persist';
import { Reducer } from 'redux';

export default (reducers: Reducer) => {
  const persistedReducer = persistReducer(
    {
      key: 'bankofmars',
      storage,
      whitelist: ['auth']
    },
    reducers
  );

  return persistedReducer;
};
