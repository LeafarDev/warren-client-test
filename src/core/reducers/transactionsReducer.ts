import * as actions from '../actionTypes/transactionsActionTypes';

export interface ITransaction {
  id: string | null
  sourceAccountNumber: string
  targetAccountNumber: string | null
  amount: number
  type: 'withdraw' | 'deposit' | 'payment'
  createdAt: string | null
  updatedAt: string | null
}

export interface TransactionsState {
  transactions: ITransaction[];
  lastCreatedTransaction: ITransaction | null;
  currentCreatingTransactionType: 'withdraw' | 'deposit' | 'payment';
}

const initialState: TransactionsState = {
  transactions: [],
  currentCreatingTransactionType: 'deposit',
  lastCreatedTransaction: null
};

export default function transactionsReducer(
  state: TransactionsState = initialState,
  action: actions.TransactionsAction
): TransactionsState {
  switch (action.type) {
    case actions.CURRENT_CREATING_TRANSACTION_TYPE:
      return {
        ...state,
        currentCreatingTransactionType: action.currentCreatingTransactionType
      };
    case actions.SAVE_TRANSACTION_SUCCESS:
      return {
        ...state,
        lastCreatedTransaction: action.transaction
      };
    case actions.GET_TRANSACTIONS_SUCCESS:
      return {
        ...state,
        transactions: action.transactions
      };
    default:
      return state;
  }
}
