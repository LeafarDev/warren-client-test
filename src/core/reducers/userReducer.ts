import * as actions from '../actionTypes/userActionTypes';

export interface IAccount {
  id: string
  accountNumber: string
  balance: number
  createdAt: string
  updatedAt: string | null
}

export interface IUser {
  id: string
  name: string
  email: string
  accountId: string
  createdAt: string
  updatedAt: string | null
  account: IAccount
}

export interface UserState {
  user: IUser
}

const initialState: UserState = {
  user: {
    id: '',
    name: '',
    email: '',
    accountId: '',
    createdAt: '',
    updatedAt: null,
    account: {
      id: '',
      accountNumber: '',
      balance: 0,
      createdAt: '',
      updatedAt: ''
    }
  }
};

export default function userReducer(
  state: UserState = initialState,
  action: actions.UserAction
): UserState {
  switch (action.type) {
    case actions.SET_USER:
    case actions.GET_USER_SUCCESS:
      return {
        user: action.user
      };
    default:
      return state;
  }
}
