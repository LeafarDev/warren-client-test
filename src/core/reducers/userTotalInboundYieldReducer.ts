import * as actions from '../actionTypes/userTotalInboundYieldActionTypes';

export interface IUserTotalInboundYield {
  total: number
}

export interface UserTotalInboundYieldState {
  userTotalInboundYield: IUserTotalInboundYield;
}

const initialState: UserTotalInboundYieldState = {
  userTotalInboundYield: {
    total: 0
  }
};

export default function userTotalInboundYieldReducer(
  state: UserTotalInboundYieldState = initialState,
  action: actions.UserTotalInboundYieldTypesAction
): UserTotalInboundYieldState {
  switch (action.type) {
    case actions.GET_USER_TOTAL_INBOUND_YIELD_SUCCESS:
      return {
        userTotalInboundYield: action.userTotalInboundYield
      };
    default:
      return state;
  }
}
