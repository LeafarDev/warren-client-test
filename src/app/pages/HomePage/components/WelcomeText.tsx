import React from 'react';
import { WelcomeGrid } from '../styles';
import { Typography } from '@material-ui/core';
import { IUser } from '../../../../core/reducers/userReducer';

interface Props {
  user: IUser,
}

export const WelcomeText: React.FC<Props> = props => {
  const {
    user
  } = props;

  return (<WelcomeGrid item xs={12}>
    <Typography
      variant="h5"> {`Welcome ${user.name} ! Account: ${user.account.accountNumber}`} </Typography>
  </WelcomeGrid>);
};
