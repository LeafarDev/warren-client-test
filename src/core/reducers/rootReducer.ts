import { combineReducers } from 'redux';
import isLoadingReducer from './isLoadingReducer';
import errorReducer from './errorReducer';
import authReducer from './authReducer';
import transactionsReducer from './transactionsReducer';
import userReducer from './userReducer';
import userTotalInboundYieldReducer from './userTotalInboundYieldReducer';

const rootReducer = combineReducers({
  userTotalInboundYield: userTotalInboundYieldReducer,
  transactions: transactionsReducer,
  auth: authReducer,
  user: userReducer,
  isLoading: isLoadingReducer,
  error: errorReducer
});

export type AppState = ReturnType<typeof rootReducer>;

export default rootReducer;
