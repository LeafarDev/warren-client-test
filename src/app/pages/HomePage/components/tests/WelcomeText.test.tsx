import React from 'react';
import { WelcomeText } from '../WelcomeText';
import { render, screen } from '@testing-library/react';
import { IUser } from '../../../../../core/reducers/userReducer';

describe('<WelcomeText />', () => {
  const user: IUser = {
    'id': '1aafa847-b0b4-42c9-929f-61cc75b975a5',
    'name': 'Charles Marciano',
    'email': 'charles.marciano45@gmail.com',
    'accountId': '1539c5cc-8978-41c2-9339-1e34778f5b7d',
    'createdAt': '2021-06-02T01:42:01.987Z',
    'updatedAt': null,
    'account': {
      'id': '1539c5cc-8978-41c2-9339-1e34778f5b7d',
      'accountNumber': '12345678',
      'balance': 0.0056,
      'createdAt': '2021-06-02T01:42:01.766Z',
      'updatedAt': '2021-06-02T21:42:02.128Z'
    }
  };

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild }
    } = render(<WelcomeText user={user}/>);

    expect(firstChild)
      .toMatchSnapshot();
  });

  it('should render the welcome message', () => {

    render(<WelcomeText user={user}/>);

    expect(screen.getByText('Welcome Charles Marciano ! Account: 12345678'))
      .toBeInTheDocument();
  });
});
