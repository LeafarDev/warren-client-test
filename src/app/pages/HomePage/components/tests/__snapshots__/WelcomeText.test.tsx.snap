// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeText /> should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root sc-gtsrHT jtaqrK MuiGrid-item MuiGrid-grid-xs-12"
>
  <h5
    class="MuiTypography-root MuiTypography-h5"
  >
     
    Welcome Charles Marciano ! Account: 12345678
     
  </h5>
</div>
`;
